### PR TITLE
Improve login page with JS

### PR DESCRIPTION
**Pull request by nono**
Makes #303 less a hack

### CODE REVIEW

File: assets/scripts/async-submit.js [hunk 1]
```diff
@@ -20,6 +20,7 @@
     }
 
     errorPanel.innerHTML = `<p>${error}</p>`
+    submitButton.removeAttribute('disabled')
```
Comment (gregorylegarec) [inline assets/scripts/async-submit.js]: 👍

File: web/auth/auth.go [hunk 2]
```diff
@@ -97,34 +94,50 @@ func loginForm(c echo.Context) error {
...
-		"Redirect":          redirect,
+		"PublicName":       publicName,
+		"CredentialsError": nil,
+		"Redirect":         redirect,
 	})
 }
 
 func login(c echo.Context) error {
 	instance := middlewares.GetInstance(c)
+	wantsJSON := c.Request().Header.Get("Accept") == "application/json"
```
Comment (gregorylegarec) [inline web/auth/auth.go]: :+1:

File: assets/scripts/async-submit.js [hunk 3]
```diff
@@ -31,25 +31,27 @@
...
-      } else {
-        throw new Error('The credentials you entered are incorrect, please try again.')
-      }
+      const loginSuccess = response.status < 400
+      response.json().then((body) => {
+        if (loginSuccess) {
+          submitButton.innerHTML = '<i class="fa fa-check"></i>'
+          submitButton.classList.add('btn-success')
+          if (body.redirect) {
+            window.location = body.redirect
```
Comment (gregorylegarec) [inline assets/scripts/async-submit.js]: Faaaaaar better 🎉

File: assets/scripts/async-submit.js [hunk 4]
```diff
@@ -31,25 +31,27 @@
...
+          } else {
+            form.submit()
```
Comment (gregorylegarec) [inline assets/scripts/async-submit.js]: What are the cases where the `form.submit()` is called ?

File: pkg/apps/apps.go [hunk 5]
```diff
@@ -43,7 +43,7 @@ const (
 const (
 	OnboardingSlug = "onboarding"
 	HomeSlug       = "home"
-	FilesSlug       = "files"
+	FilesSlug      = "files"
```
Comment (gregorylegarec) [inline pkg/apps/apps.go]: So sorry about this one. :(

(Great example of what kind of typos aligned values may cause, I am not a big fan but I swear it was an accident)

**Comment by gregorylegarec** [comment]
Do we wait for @m4dz review ? I think he is pretty much busy today. If we merge it now, it could still be validated by the product team today.

File: pkg/apps/apps.go (hunk 5 above)
Comment (nono) [inline pkg/apps/apps.go]: The golang developers have gofmt called when they save a file, and it does this stuff for us. It is really cool in golang to not have to think about code style nitpicks like how many spaces for indentation, and just let gofmt does this work for us!

To be honest, we don't use gofmt, but goimports. It is the same, but it also adds the imports automatically.

**Comment by nono** [comment]
Let's merge!

File: pkg/apps/apps.go (hunk 5 above)
Comment (gregorylegarec) [inline pkg/apps/apps.go]: Thanks for the tips !

**Pull request merged**